### PR TITLE
HIVE-28514: Iceberg: Fix checkstyle file to adapt new checkstyle version

### PR DESCRIPTION
**Pull request by zhangbutao**
### What changes were proposed in this pull request?

Upgrading maven maven-checkstyle-plugin to latest 3.5.0 and change checkstyle file & some class to adapt the higher version checkstyle rules.

### Why are the changes needed?

See details HIVE-28514 & HIVE-28499.

BTW, the new CheckStyle version has some extra check rules. I have fixed them in this change.
```
[ERROR] hive/iceberg/iceberg-catalog/src/main/java/org/apache/iceberg/hive/HiveClientPool.java:91:46: Unnecessary parentheses around expression. [UnnecessaryParentheses]
[ERROR] hive/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java:1546: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
[ERROR] hive/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java:1561: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
[ERROR] hive/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java:1748: Block tags have to appear in the order '[@param, @return, @throws, @deprecated]'. [AtclauseOrder]
[ERROR] hive/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java:1749: Block tags have to appear in the order '[@param, @return, @throws, @deprecated]'. [AtclauseOrder]
[ERROR] hive/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java:1751: Block tags have to appear in the order '[@param, @return, @throws, @deprecated]'. [AtclauseOrder]
[ERROR] hive/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java:411:15: Unnecessary parentheses around expression. [UnnecessaryParentheses]
[ERROR] hive/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java:412:15: Unnecessary parentheses around expression. [UnnecessaryParentheses]
[ERROR] hive/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java:434:15: Unnecessary parentheses around expression. [UnnecessaryParentheses]
[ERROR] hive/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java:435:15: Unnecessary parentheses around expression. [UnnecessaryParentheses]
[ERROR] hive/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java:1022:10: Unnecessary parentheses around expression. [UnnecessaryParentheses]
[ERROR] hive/iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergSelects.java:121:11: Unnecessary parentheses around expression. [UnnecessaryParentheses]
[ERROR] hive/iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergSelects.java:148:11: Unnecessary parentheses around expression. [UnnecessaryParentheses]
```

### Does this PR introduce _any_ user-facing change?

No

### Is the change a dependency upgrade?

No

### How was this patch tested?

mvn build Hive Iceberg handler.

### CODE REVIEW

File: iceberg/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -448,7 +448,7 @@
           <property name="max" value="12"/>
         </module>
         <module name="JavadocMethod"> <!-- Java Style Guide: Where Javadoc is used -->
-            <property name="scope" value="public"/>
+            <property name="accessModifiers" value="public"/>
```
Comment (zhangbutao) [inline iceberg/checkstyle/checkstyle.xml:451]: After this change, you can use the Intellij 2024 with higher CheckStyle-IDEA plugin version 5.94.1:
idea2024-Checkstyle-IDEA

idea2024-hive-ice-check

**Comment by Aggarwal-Raghav** [comment]
LGTM+1 (waiting for test result),
This will help with the CVE for velocity 1.7, getting downloaded as part of 3.1.2 plugin version :-)

**Comment by Aggarwal-Raghav** [comment]
**_nit:_** Can we handle the [WARNING]?

```
[INFO] --------------------< org.apache.hive:hive-iceberg >--------------------
[INFO] Building Hive Iceberg Modules 4.1.0-SNAPSHOT                       [1/6]
[INFO]   from pom.xml
[INFO] --------------------------------[ pom ]---------------------------------
[WARNING] Parameter 'encoding' is unknown for plugin 'maven-checkstyle-plugin:3.5.0:check (validate)'
[INFO]
[INFO] --- clean:3.1.0:clean (default-clean) @ hive-iceberg ---
[INFO] Deleting /Users/raghav/Desktop/apache/hive/master/iceberg/target
[INFO] Deleting /Users/raghav/Desktop/apache/hive/master/iceberg (includes = [datanucleus.log, derby.log], excludes = [])
```

**Review by Aggarwal-Raghav — APPROVED** [review]

**Review by ayushtkn — APPROVED** [review]
LGTM

**Pull request merged**